ggml.c

```diff
@@ -505,18 +505,29 @@ static inline __m256 sum_i16_pairs_float(const __m256i x) {
 
 // multiply int8_t, add results pairwise twice and return as float vector
 static inline __m256 mul_sum_i8_pairs_float(const __m256i x, const __m256i y) {
+#if __AVXVNNIINT8__
+    const __m256i zero = _mm256_setzero_si256();
+    const __m256i summed_pairs = _mm256_dpbssd_epi32(zero, x, y);
+    return _mm256_cvtepi32_ps(summed_pairs);
+#else
```

Comment:
As far as I'm aware, there is no hardware out there supporting `AVX-VNNI-INT8` yet, so I don't think it's a good idea to use `_mm256_dpbssd_epi32()` here (since we cannot test it).

What we can use is `AVX-VNNI`. It is present on already existing Intel CPUs starting from Alder Lake (12th gen) and includes `_mm256_dpbusd_epi32()`. The difference is the left operand should be unsigned, so you have to keep `_mm256_sign_epi8(...)` and only replace the `_mm256_maddubs_epi16() + sum_i16_pairs_float()` pair with `_mm256_dpbusd_epi32()`.

Essentially, this would be a backport of what I did with AVX-512 (zmm registers) to AVX-VNNI (ymm registers), which was proposed by @ultoris [here](https://github.com/ggerganov/llama.cpp/pull/933#issuecomment-1509865053).

Maybe we should open a separate PR for the VNNI optimization? It might speed-up the quantized dot product, which is on the hot code path, so it's nice to get some performance measurements independently of the `packNibbles()` optimization.

Reply:
> What we can use is `AVX-VNNI`. It is present on already existing Intel CPUs starting from Alder Lake (12th gen) and includes `_mm256_dpbusd_epi32()`. The difference is the left operand should be unsigned, so you have to keep `_mm256_sign_epi8(...)` and only replace the `_mm256_maddubs_epi16() + sum_i16_pairs_float()` pair with `_mm256_dpbusd_epi32()`.

Good suggestion! I have changed `mul_sum_i8_pairs_float` to use AVX_VNNI, I also think the `packNibbles` does not affect the inference speed, so I think it is ok to measure inference performance in this PR.